src/main/java/seedu/toluist/controller/commons/IndexTokenizer.java

```diff
@@ -0,0 +1,89 @@
+package seedu.toluist.controller.commons;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import seedu.toluist.commons.util.StringUtil;
+
+/**
+ * Tokenize string of indexes into a list of integer index.
+ */
+public class IndexTokenizer {
+    private static final String HYPHEN = "-";
+
+    /**
+     * Returns the index if indexToken holds a valid index number, else returns -1
+     * @param indexToken the string holding the index number
+     */
+    public static int getIndex(String indexToken) {
+        return indexToken == null ? -1 : Integer.parseInt(indexToken) - 1;
+    }
+
+    /**
+     * Splits a string of indexes into a list.
+     *   Examples:
+     *   splitIndexes(" - 3",           8) -> [1, 2, 3]
+     *   splitIndexes(" 3 -",           8) -> [3, 4, 5, 6, 7, 8]
+     *   splitIndexes("3 4,  5",        8) -> [3, 4, 5]
+     *   splitIndexes("3 - 5",          8) -> [3, 4, 5]
+     *   splitIndexes("- 3, 5, 7 - 12", 8) -> [1, 2, 3, 5, 7, 8]
+     * This function is robust enough to handle excessive white spaces (" ") and commas (",").
+     * @param stringIndexes, the unprocessed string of indexes from user's input
+     * @param maxIndex, the maximum possible index number
+     * @return a list of integer containing all valid indexes
+     */
+    public static List<Integer> splitIndexes(String stringIndexes, int maxIndex) {
```

Comment:
Let's go with not possible right now since i terminate the moment i process an invalid index. So we delete whatever that is still valid and stop at that. 
Feedback message we can just do something like
"Delete task 1"
"Delete task 3" 
"Delete task 5" 
etcetc (one for each)